numpy/core/src/multiarray/calculation.c

```diff
@@ -87,9 +88,12 @@ _PyArray_ArgMinMaxCommon(PyArrayObject *op,
         op = ap;
     }
 
-    /* Will get native-byte order contiguous copy. */
-    ap = (PyArrayObject *)PyArray_ContiguousFromAny((PyObject *)op,
-                                  PyArray_DESCR(op)->type_num, 1, 0);
+    // Will get native-byte order contiguous copy.
+    ap = (PyArrayObject *)PyArray_FromArray(
+        op,
+        PyArray_DescrNew(NPY_DT_CALL_ensure_canonical(PyArray_DESCR(op))),
```

Comment:
Thanks, unfortunately, the `NPY_DT_CALL_ensure_canonical` can fail.  Also don't use `PyArray_DescrNew`, `NPY_DT_CALL_ensure_canonical` already returns a new reference.

Reply:
Thanks, I added a guard against a bad call to `ensure_canonical`, and removed the call to `PyArray_DescrNew`.

I know a dtype is canonical if its associated array has native byte order - but what exactly is `ensure_canonical` meant to do? I've looked everywhere I can think of and I've not found documentation, except a little bit in NEP42. The implementations that we have for the user dtypes just increment and return the descriptor. Should these be reordering the array data if the byte order is not correct?